### ISSUE TITLE
Serverless instance returns incorrect connection string

### ISSUE THREAD

**Issue by jsmi565** [mongodb/terraform-provider-mongodbatlas#934]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform version: 1.3.5
Provider version: 1.6.0
```

### Steps to Reproduce

1. Use Serverless PrivateLink example in docs.
2. Output connection string.
3. Compare outputted connection string with connection string in Atlas console.

### Expected Behavior

Both connection strings should be the same?

### Actual Behavior

Both connection strings differ from one another.

### Debug Output 

Outputted connection string: `mongodb+srv://main.same.mongodb.net`
Atlas console connection string (for the same AWS PrivateLink): `mongodb+srv://main-pe-0.same.mongodb.net`

### Additional Context

My MongoDB setup in Go throws the following error:

`Failed to connect to MongoDB: connection() error occurred during connection handshake: handshake failure: connection(main-lb.same.mongodb.net:27017[-2]) socket was unexpectedly closed: EOF`

**Comment by jsmi565** [comment]
Double checked with support that this may be an issue.

They said:

> Please note that to connect to the Atlas cluster over private endpoint, you would be required to connect using -pe- connection string.

Current workaround is the following:

```
locals {
  mongodb_instance_id    = split(".", mongodbatlas_serverless_instance.main.connection_strings_standard_srv)[1]
  mongodb_connection_uri = "mongodb+srv://${mongodbatlas_serverless_instance.main.name}-pe-0.${local.mongodb_instance_id}.mongodb.net"
}
```

Doesn't scale with more than one PEs though.

**Comment by Zuhairahmed** [comment]
thanks @jsmi565. this is a good call out and we created an internal ticket ([INTMDB-493](https://jira.mongodb.org/browse/INTMDB-493)) to resolve. adding to our roadmap to help in enable you and future users to be able to leverage desired connection strings. given competiting priorities though, this may take several months to address. will keep you updated when issue has been resolved.

**Comment by martinstibbe** [comment]
@jsmi565 v1.6.1 release will contain this fix

**Comment by Zuhairahmed** [comment]
@jsmi565 we were actually able to squeeze this fix as part of v1.6.1 which released today which includes this fix. Feel free to upgrade Terraform Provider versions and let us know if this resolves the issue.
